test/view/broadcast/broadcast_game_screen_test.dart

```diff
@@ -0,0 +1,119 @@
+import 'package:lichess_mobile/src/network/http.dart';
+import 'package:lichess_mobile/src/network/socket.dart';
+import 'package:lichess_mobile/src/view/broadcast/broadcast_game_screen.dart';
+
+import '../../network/fake_websocket_channel.dart';
+import '../../test_helpers.dart';
+import '../../test_provider_scope.dart';
+
+final client = MockClient((request) {
+  if (request.url.path == '/api/broadcast/-/-/$_roundId') {
+    return mockResponse(
+      _roundResponse,
+      200,
+      headers: {'content-type': 'application/json; charset=utf-8'},
+    );
+  }
+  if (request.url.path == '/api/study/$_roundId/$_gameId.pgn') {
+    return mockResponse(_gameResponse, 200, headers: {'content-type': 'application/x-chess-pgn'});
+  }
+  return mockResponse('', 404);
+});
+
+void main() {
+  group('Broadcast game screen', () {
+    testWidgets('Displays broadcast game screen', variant: kPlatformVariant, (tester) async {
+      final fakeSocket = FakeWebSocketChannel();
+      final app = await makeTestProviderScopeApp(
+        tester,
+        home: const BroadcastGameScreen(
+          tournamentId: _tournamentId,
+          roundId: _roundId,
+          gameId: _gameId,
+        ),
+        overrides: [
+          lichessClientProvider.overrideWith((ref) => LichessClient(client, ref)),
+          webSocketChannelFactoryProvider.overrideWith((ref) {
+            return FakeWebSocketChannelFactory((_) => fakeSocket);
+          }),
+        ],
+      );
+
+      await tester.pumpWidget(app);
+
+      expect(find.byType(CircularProgressIndicator), findsOneWidget);
+
+      await tester.pump();
+
+      expect(find.byType(Chessboard), findsOneWidget);
+    });
+
+    testWidgets('Receives a new move of the game', variant: kPlatformVariant, (tester) async {
+      final fakeSocket = FakeWebSocketChannel();
+      final app = await makeTestProviderScopeApp(
+        tester,
+        home: const BroadcastGameScreen(
+          tournamentId: _tournamentId,
+          roundId: _roundId,
+          gameId: _gameId,
+        ),
+        overrides: [
+          lichessClientProvider.overrideWith((ref) => LichessClient(client, ref)),
+          webSocketChannelFactoryProvider.overrideWith((ref) {
+            return FakeWebSocketChannelFactory((_) => fakeSocket);
+          }),
+        ],
+      );
+
+      await tester.pumpWidget(app);
+
+      await tester.pump();
+
+      expect(find.byKey(const Key('c1-whitebishop')), findsOneWidget);
+
+      fakeSocket.addIncomingMessages([
+        r'''{"v":151,"t":"addNode","d":{"n":{"ply":23,"fen":"rnq2rk1/pp2ppbp/5np1/2P1N3/2N5/4B1P1/PP2PPKP/R2Q1R2 b - - 2 12","id":"%7","uci":"c1e3","san":"Be3","clock":359500},"p":{"chapterId":"G2LUflKg","path":")8aP19YQ(1`Y'*_b.>VF-=F=$3UE3=]O8GOF>EF1)1^]"},"d":"0S 978 TCEJNZ8 WGO 3NV 6xEILQSYZ78 !? UM 5OQZ 2V? XHP","s":false,"relayPath":"!"}}''',
+      ]);
+
+      // Two pump are required to get the move but I don't know why
```

Comment:
I think that is because how the socket is mocked. Since we're using a stream controller, the socket event is received after the next micro task. So 1 pump to wait for the event, and 1 pump to treat the event.

You can add that explanation, or better yet: make a test helper that takes the socket channel and performs the `tester.pump()`. Sth that would allow to write:

```dart
await addSocketMessage(tester, fakeSocket, '<message here>');

// wait for the controller to process the message
await tester.pump();
```

That way writing (and reading) tests will be more clear.